src/threads/ThreadState.cpp

```diff
@@ -11,6 +13,8 @@
 
 using namespace faabric::util;
 
+#define LEVEL_WAIT_TIMEOUT_MS 20000
```

Comment:
This is somewhat arbitrary, it will be hit if a thread dies and doesn't hit the wait section, but it is preferable to have a clear error after a reasonable wait rather than an infinite hang.